cli/dcoscli/marathon/main.py

```diff
@@ -229,60 +229,96 @@ def _info():
     return 0
 
 
-def _get_resource(resource):
-    """
-    :param resource: optional filename or http(s) url
-    for the application or group resource
-    :type resource: str
-    :returns: resource
-    :rtype: dict
-    """
-    if resource is not None:
-        if os.path.isfile(resource):
-            with util.open_file(resource) as resource_file:
-                return util.load_json(resource_file)
-        else:
-            try:
-                http.silence_requests_warnings()
-                req = http.get(resource)
-                if req.status_code == 200:
-                    data = b''
-                    for chunk in req.iter_content(1024):
-                        data += chunk
-                    return util.load_jsons(data.decode('utf-8'))
-                else:
-                    raise Exception
-            except Exception:
-                logger.exception('Cannot read from resource %s', resource)
+class ResourceReader(object):
+    """Encapsulates side-effecting methods for loading Marathon resources."""
+
+    @staticmethod
+    def from_filename_or_url_or_stdin(resource):
+        """
+        :param resource: optional filename or http(s) url
+        for the application or group resource
+        :type resource: str
+        :returns: resource
+        :rtype: dict
+        """
+        if resource is not None:
+            if os.path.isfile(resource):
+                with util.open_file(resource) as resource_file:
+                    return util.load_json(resource_file)
+            else:
+                try:
+                    http.silence_requests_warnings()
+                    req = http.get(resource)
+                    if req.status_code == 200:
+                        data = b''
+                        for chunk in req.iter_content(1024):
+                            data += chunk
+                        return util.load_jsons(data.decode('utf-8'))
+                    else:
+                        raise Exception
+                except Exception:
+                    logger.exception('Cannot read from resource %s', resource)
+                    raise DCOSException(
+                        "Can't read from resource: {0}.\n"
+                        "Please check that it exists.".format(resource))
+
+        # Check that stdin is not tty
+        if sys.stdin.isatty():
+            # We don't support TTY right now. In the future we will start an
+            # editor
+            raise DCOSException(
+                "We currently don't support reading from the TTY. Please "
+                "specify an application JSON.\n"
+                "E.g.: dcos marathon app add < app_resource.json")
+
+        return util.load_json(sys.stdin)
+
+    @staticmethod
+    def from_properties_or_stdin(properties):
```

Comment:
`get_properties` makes me think the return value will be the properties, but it will actually be the resource generated from the `properties` argument. How about `get_resource_from_properties` or `get_from_properties`?